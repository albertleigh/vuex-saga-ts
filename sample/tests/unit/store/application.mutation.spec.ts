import { MutationsActions, mutations } from '@/store/application/mutations';


describe('Application mutations',()=>{

    describe('act creators',()=>{

        it('ready',()=>{
            expect(''+MutationsActions.ready).toMatchSnapshot('application_mut_act_ready_name');
            expect(MutationsActions.ready()).toMatchSnapshot('application_mut_act_ready');
        })

    })

    describe('mutator',()=>{

        it('ready',()=>{
            const initSate = {
                loading: true,
            };
            expect(initSate).toMatchSnapshot('before ready mutator');
            mutations[''+MutationsActions.ready](initSate);
            expect(initSate).toMatchSnapshot('after ready mutator');
        })

    })

})
