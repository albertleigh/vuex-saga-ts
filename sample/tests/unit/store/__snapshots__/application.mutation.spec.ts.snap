// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application mutations act creators ready: application_mut_act_ready 1`] = `
Object {
  "payload": Object {},
  "type": "APPLICATION_READY",
}
`;

exports[`Application mutations act creators ready: application_mut_act_ready_name 1`] = `"APPLICATION_READY"`;

exports[`Application mutations mutator ready: after ready mutator 1`] = `
Object {
  "loading": false,
}
`;

exports[`Application mutations mutator ready: before ready mutator 1`] = `
Object {
  "loading": true,
}
`;
